### PR TITLE
Interactive code safety review

### PR DESCRIPTION
**Pull request by suezhou**
@timgentry asked me to add the ability to interactively code review, prompting to flag code as safe, rather than forcing the reviewer to run a separate rake task for each file. I have implemented this here as a new rake task `audit:review`.

### CODE REVIEW

File: lib/tasks/audit_code.rake [hunk 1]
```diff
@@ -478,6 +467,43 @@ Usage:
     flag_file_as_safe(release, ENV['reviewed_by'], ENV['comments'], ENV['file'])
   end
 
+  desc "Review safety of selected source file.
+Usage:
+  Review a source file:   #{rake_cmd} audit:review file=f [comments=...]"
+  task(:review) do
+    fname = ENV['file']
+    if fname.nil?
```
Comment (joshpencheon) [inline lib/tasks/audit_code.rake]: Unless I'm missing something, this still requires you to run `rake audit:review` for each file individually, which would seem to defeat the point a little?

File: lib/tasks/audit_code.rake [hunk 2]
```diff
@@ -403,6 +373,17 @@ def get_release
   release
 end
 
+def get_reviewer(prompt = false)
+  reviewer = ENV['reviewed_by']
+  if reviewer.nil?
+    require 'rugged'
+    repo = Rugged::Repository.new('.')
```
Comment (joshpencheon) [inline lib/tasks/audit_code.rake]: What's the behaviour here is the user is in an `svn` working copy? We don't want it to crash...

File: lib/tasks/audit_code.rake (hunk 1 above)
Comment (suezhou) [reply lib/tasks/audit_code.rake]: With existing tasks, you need to run the following command to flag a file safe, and the fact of long hash value force you to copy and paste the code.
`bundle exec rake audit:safe release=HASH_VALUE file=FILE_NAME reviewed_by=USER comments=""`

the new task require you to run the code `bundle exec rake audit:review file=FILE_NAME`, where file name can be completed by pressing TAB key, and you can then review and flag the file safe or not using the command prompt.

The main difference is to save you from repeated copy and paste.

File: lib/tasks/audit_code.rake (hunk 2 above)
Comment (suezhou) [reply lib/tasks/audit_code.rake]: I will check against a svn repository to get an answer.

File: lib/tasks/audit_code.rake (hunk 1 above)
Comment (timgentry) [reply lib/tasks/audit_code.rake]: This just isn't doing what I intended. Rather than tweak this PR further I suggest we look at this again.

**Pull request closed without merging**